### ISSUE TITLE
PHP fatal error if the requested vocabulary id is not found

### ISSUE THREAD

**Issue by kouralex** [NatLibFi/Skosmos#1170]
Whilst looking at apache2 error log, I noticed that Skosmos throws uncaught PHP fatal errors if the requested vocabulary id is not found. See the following call stack:

```PHP Fatal error:  Uncaught ValueError: Vocabulary id 'xyz' not found in configuration. in model/Model.php:420
Stack trace:
#0 model/Request.php(246): Model->getVocabulary('xyz')
#1 index.php(51): Request->setVocab('xyz')
#2 {main}
thrown in model/Model.php on line 420
```

These should be caught and handled appropriately.

**Comment by hekl** [comment]
I have something like this, these errors are thrown almost every minute. Sometimes with the last referer statement, mostly not. Everything seems fine, except that the loading of the last section of equivalent concepts seems to stall. Watch it: https://classificaties.cbs.nl/geo/def/gemeenten/GM0359

`PHP Fatal error:  Uncaught Error: Class 'ValueError' not found in /var/www/Skosmos/model/Model.php:420\nStack trace:\n#0 /var/www/Skosmos/model/Request.php(246): Model->getVocabulary()\n#1 /var/www/Skosmos/index.php(51): Request->setVocab()\n#2 {main}\n  thrown in /var/www/Skosmos/model/Model.php on line 420, referer: http://classificaties.cbs.nl/sess-bin/login_session.cgi
`

**Comment by osma** [comment]
@hekl It seems that your PHP installation doesn't know about the ValueError class that is used in some parts of Skosmos code since PR #1127 which, among other things, upgraded PHPUnit to version 8. I started investigating why this could happen, and it turns out ValueError was only introduced in PHP 8. But it is also defined by [symfony/polyfill-php80](https://github.com/symfony/polyfill-php80) which Skosmos depends on indirectly through twig-gettext-extractor -- but only if development dependencies are installed! So that explains why you are getting this error (you've used `composer install --no-dev` as instructed in the InstallTutorial) while Skosmos developers usually have the development dependencies, obviously :)

I will open a separate bug report about this "missing ValueError" problem.

**Comment by osma** [comment]
This was mostly fixed in PR #1215 (and the separate but similar problem with the ValueError class, encountered by @hekl, was fixed in PR #1200).

However, looking at Finto server logs, there are still problems in the REST API; uncaught ValueErrors appear for REST API calls with an unknown vocabulary ID.
